pkg/controller/expectations.go

```diff
@@ -324,12 +324,13 @@ func (u *UIDTrackingControllerExpectations) ExpectDeletions(rcKey string, delete
 	}
 	glog.V(4).Infof("Controller %v waiting on deletions for: %+v", rcKey, deletedKeys)
 	if err := u.uidStore.Add(&UIDSet{expectedUIDs, rcKey}); err != nil {
-		return err
+		// can't happen as long we don't add something other than UIDSet
```

Comment:
If the error can't happen, why not just call `u.uidStore.Add(&UIDSet{expectedUIDs, rcKey})` and ignore the returned value?

Reply:
That would be an unhandled error which would be then reported by gosec. I prefer to explicitly write this rather than #nosec.

Reply:
Makes sense thx :+1:

Reply:
should we at least log there error for debug purposes? By completely ignoring all errors I think it might make it hard for a program to debug if something goes awry.

Reply:
I can add that but the error can only happen if the `KeyFunc` returns it. The error then can only be made by us adding the wrong type. As this is only in the scope of this package(even more strictly in scope of the `ControllerExpectations` ) I think with unit tests and reviews its hard to get this error. WDYT?

Reply:
I'm on the same side as Ashley. My take on unexpected errors is that they never should go unnoticed. So I'd rather panic on it.